plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/ADBInfo.java

```diff
@@ -0,0 +1,99 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.tools.openshift.cdk.server.core.internal.listeners;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.tools.openshift.cdk.server.core.internal.CDKConstantUtility;
+import org.jboss.tools.openshift.cdk.server.core.internal.CDKConstants;
+import org.jboss.tools.openshift.cdk.server.core.internal.CDKCoreActivator;
+import org.jboss.tools.openshift.cdk.server.core.internal.adapter.CDKServer;
+import org.jboss.tools.openshift.cdk.server.core.internal.adapter.controllers.VagrantLaunchUtility;
+
+public class ADBInfo {
+	
+	public static final String SHARED_INFO_KEY = "cdk.sharedinfo.adbInfo";
+	
+	int openshiftPort = 8443;
+	String openshiftHost = "https://10.1.2.2";
+	
+	HashMap<String,String> env;
+	public ADBInfo(HashMap<String,String> env) throws URISyntaxException {
+		this.env = env;
+		String dockerHost = env.get("DOCKER_HOST");
+		URI url = new URI(dockerHost);
+		String h = url.getHost();
+		openshiftHost = "https://" + h;
```

Comment:
We can't.  the scheme returned from adbinfo is tcp://  (or sometimes https on other operating systems) so it's inconsistant enough but we know that we need https for openshift.